reactivesearch/src/custom/AlgoPicker.js

```diff
@@ -1,55 +1,168 @@
-import React, { Component } from 'react';
-import Title from '../styles/Title';
-import Container from '../styles/Container';
-import {getClassName} from '@appbaseio/reactivecore/lib/utils/helper';
-
+import React, { Component } from "react";
+import Title from "../styles/Title";
+import Container from "../styles/Container";
+import { getClassName } from "@appbaseio/reactivecore/lib/utils/helper";
+import axios from "axios";
 
 class AlgoPicker extends Component {
   state = {
-    algo: 'keyword',
-    selectedValue: 'keyword',
+    selectedAlgo: "keyword",
     showTextBox: false,
+    showConfigDropdown: false,
     conf_a: undefined,
-    conf_b: undefined
+    conf_b: undefined,
+    selectedConfig: "",
+    availableConfigs: [],
+    configError: null,
+  };
+
+  // Wait until the component is mounted to fetch the search configurations
+  componentDidMount() {
+    this.fetchSearchConfigurations();
+  }
+
+  fetchSearchConfigurations = async () => {
+    try {
+      const serverUrl = this.props.eventServer || "http://localhost:9090";
```

Comment:
do we need the http?   I think we should figure out how to have it always be set...  otehrwise, what happens when we deploy on our chorus server?